llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -2553,6 +2553,34 @@ void CombinerHelper::applyCombineTruncOfExt(
   MI.eraseFromParent();
 }
 
+bool CombinerHelper::matchCombineShlOfAnd(MachineInstr &MI, Register &Reg) {
+  assert(MI.getOpcode() == TargetOpcode::G_SHL && "Expected a G_SHL");
+  Register SrcReg = MI.getOperand(1).getReg();
+  Register ShftReg = MI.getOperand(2).getReg();
+
+  const auto ShftVal = getIConstantVRegSExtVal(ShftReg, MRI);
+  if (!ShftVal)
+    return false;
+  MachineInstr *SrcMI = MRI.getVRegDef(SrcReg);
```

Comment:
This is unsafe, getVRegDef can return nullptr (https://llvm.org/doxygen/MachineRegisterInfo_8cpp_source.html#l00407)
Add a check to make sure SrcMI is not a nullptr.